### PR TITLE
Added subdomain takeover module

### LINKED ISSUE #309: Subdomain takeovers
The current version of Nettacker do not checks for subdomain takeover vulnerability.

_________________
**OS**: `Linux`

**OS Version**: `Ubuntu 18.04`

**Python Version**: `3.6.9`

### PR DESCRIPTION
**Pull request by aman566**
Added subdomain takeover module. Fixed #309. Just use.
Using `-s` switch with the following command help in finding subdomain takeover vulnerabilities:
`python3 nettacker.py -i github.io -s -m subdomain_takeover_vuln`
#### Checklist
- [x] I have followed the [Contributor Guidelines](https://github.com/zdresearch/OWASP-Nettacker/wiki/Developers#contribution-guidelines).
- [x] The code has been thoroughly tested in my local development environment with flake8 and pylint.
- [x] The code is both Python 2 and Python 3 compatible.
- [x] The code follows the PEP8 styling guidelines with 4 spaces indentation.
- [x] This Pull Request relates to only one issue or only one feature
- [x] I have referenced the coresspoinding issue number in my commit message
- [ ] I have added the relevant documentation.
- [x] My branch is up-to-date with the Upstream master branch.

#### Your development environment
- OS: `Linux`
- OS Version: `Ubuntu 18.04`
- Python Version: `3.6`

Resource Used: https://github.com/EdOverflow/can-i-take-over-xyz

### CODE REVIEW

**Comment by Ali-Razmjoo** [comment]
@securestep9 this sounds like a very good module to have, if you have the time review and merge (prioritize this)

File: lib/vuln/subdomain_takeover/engine.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+        "sub_takeover_smartjobboard": [
+            "This job board website is either expired or its domain name is invalid."
+        ],
+        "sub_takeover_strikingly": ["page not found"],
+        "sub_takeover_agilecrm": ["Sorry, this page is no longer available."],
+        "sub_takeover_gemfury": ["404: This page could not be found."],
+        "sub_takeover_getresponse": ["With GetResponse Landing Pages, lead generation has never been easier"],
+        "sub_takeover_juzzhr": ["This account no longer active"],
+        "sub_takeover_landingi": ["The page you are looking for is not found"],
+        "sub_takeover_smugmug": ["{\"text\":\"Page Not Found\""]
```
Comment (securestep9) [inline lib/vuln/subdomain_takeover/engine.py:129]: @aman566 have the patterns loaded in a separate wordlist in  lib/payload/wordlists similar to how you implemented dir_scan - also check other subdomain take over patterns defined in nuclei-templates : https://github.com/projectdiscovery/nuclei-templates/blob/master/subdomain-takeover/detect-all-takeovers.yaml

https://github.com/projectdiscovery/nuclei-templates/blob/master/subdomain-takeover/s3-subtakeover.yaml

File: lib/vuln/subdomain_takeover/engine.py [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+        __log_into_file(thread_tmp_filename, "w", "0", language)
+        data = json.dumps(
+            {
+                "HOST": target,
+                "USERNAME": "",
+                "PASSWORD": "",
+                "PORT": port,
+                "TYPE": "subomain_takeover_vuln",
+                "DESCRIPTION": messages(language, "vulnerable").format(
+                    "Subdomain Takeover Vulnerability found which will allow an adversary to claim and take control of the victim's subdomain."
```
Comment (securestep9) [inline lib/vuln/subdomain_takeover/engine.py:254]: It is not 100% certain that takeover vulnerability  so please change the text to "Potential Subdomain Takeover Vulnerability found" and list the pattern name found (e.g. heroku /uservoice  etc...)

File: lib/vuln/subdomain_takeover/engine.py (hunk 1 above)
Comment (aman566) [reply lib/vuln/subdomain_takeover/engine.py]: Okay sir,
Done the changes as suggested :)

File: lib/vuln/subdomain_takeover/engine.py (hunk 2 above)
Comment (aman566) [reply lib/vuln/subdomain_takeover/engine.py]: Changed as suggested

**Comment by securestep9** [comment]
@aman566 for some reason this module does not work under python2 - please check.

 https://travis-ci.com/github/OWASP/Nettacker/jobs/468282981#L1052

**Comment by aman566** [comment]
Sir I have fixed the error. Please review it

**Comment by securestep9** [comment]
@aman566 still getting Unicode error in Python2:

```
/lib/vuln/subdomain_takeover/engine.py", line 105, in sub_takeover
    value[0].lower() in req.text.lower()
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 84: ordinal not in range(128)
```

to reproduce try:
`
python nettacker.py -m subdomain_takeover_vuln -i owasp.github.io`

**Pull request merged**